src/main/java/pl/simpleascoding/tutoringplatform/service/user/UserServiceImpl.java

```diff
@@ -110,21 +111,53 @@ public String confirmUserRegistration(String tokenValue) {
      * @param username         Username of the user
      * @return The status of the operation
      */
+
     @Override
     @Transactional
-    public String changeUserPassword(ChangeUserPasswordDTO newPasswordsData, String username) {
+    public String changeUserPassword(ChangeUserPasswordDTO newPasswordsData, String username, String rootUrl) {
         User userEntity = userRepository.findUserByUsername(username)
                 .orElseThrow(() -> new UsernameNotFoundException(username));
 
         if (isChangeAllowed(userEntity.getPassword(), newPasswordsData)) {
-            userEntity.setPassword(passwordEncoder.encode(newPasswordsData.newPassword()));
+
+            Token token = new Token(TokenType.CHANGE_PASSWORD_CONFIRMATION);
+            String password = newPasswordsData.newPassword();
+            token.setData(passwordEncoder.encode(password));
+            userEntity.getTokens().add(token);
+
+            SimpleMailMessage message = new SimpleMailMessage();
+            message.setTo(userEntity.getEmail());
+            message.setSubject(PASSWORD_CHANGE_CONFIRMATION_MAIL_SUBJECT);
+            message.setText(String.format(PASSWORD_CHANGE_CONFIRMATION_MAIL_TEXT, userEntity.getName(), String.format(PASSWORD_CHANGE_CONFIRMATION_LINK, rootUrl, token.getValue())));
+
+            mailSender.send(message);
 
             return HttpStatus.OK.getReasonPhrase();
```

Comment:
Instead of returning **HttpStatus** - please use **RscpDTO** with **RscpStatus** inside of it.